internal.lua

```diff
@@ -9,78 +10,71 @@ local F = minetest.formspec_escape
 -- This is a game engine bug, and in the anticipation that it might be
 -- fixed some day we don't want to rely on it.  So for safety we apply
 -- an encoding that avoids all formspec metacharacters.
-function unified_inventory.mangle_for_formspec(str)
+
+function ui.mangle_for_formspec(str)
 	return string.gsub(str, "([^A-Za-z0-9])", function (c) return string.format("_%d_", string.byte(c)) end)
 end
-function unified_inventory.demangle_for_formspec(str)
+function ui.demangle_for_formspec(str)
 	return string.gsub(str, "_([0-9]+)_", function (v) return string.char(v) end)
 end
 
-function unified_inventory.get_per_player_formspec(player_name)
-	local lite = unified_inventory.lite_mode and not minetest.check_player_privs(player_name, {ui_full=true})
 
-	local ui = {}
-	ui.pagecols = unified_inventory.pagecols
-	ui.pagerows = unified_inventory.pagerows
-	ui.page_y = unified_inventory.page_y
-	ui.formspec_y = unified_inventory.formspec_y
-	ui.main_button_x = unified_inventory.main_button_x
-	ui.main_button_y = unified_inventory.main_button_y
-	ui.craft_result_x = unified_inventory.craft_result_x
-	ui.craft_result_y = unified_inventory.craft_result_y
-	ui.form_header_y = unified_inventory.form_header_y
+function ui.get_per_player_formspec(player_name)
+	local lite = ui.lite_mode and not minetest.check_player_privs(player_name, {ui_full=true})
+
+	local style = ui.style_full
```

Comment:
```suggestion
	local style = table.copy(draw_lite_mode and ui.style_full or ui.style_lite)
```
This eliminates corruption issues caused by per-player table modifications (by reference). Also shortens the code a bit.

EDIT: Renamed `lite` to `draw_lite_mode` to be consistent with the caller function's variable names.

Reply:
Actually, need to push that commit out.  it broke things kinda badly.

Reply:
Make that two commits.  You inverted the meaning of the lite flag. :smile:

Reply:
This is fixed now.